bin/string_encoder

```diff
@@ -5,7 +5,8 @@
 # $ cat some_file | string_encoder
 
 while input = gets
-  if input.to_s.valid_encoding?
+  input = input.to_s.force_encoding('UTF-8')
+  if input.valid_encoding?
```

Comment:
This **if** condition will always pass because of the force_encoding. We should be simply replacing **encode** with **force_encoding**.

Reply:
Nopes, when we replace *encode* with *force_encoding* the invalid byte sequence wouldn't be replaced with the `?` and secondly, the current(existing) sequence is just about right and doing the force encoding helps to retain the original data (for valid encodings)

Reply:
Agreed